public/components/custom_panels/forms/add_new_visualizations.tsx

```diff
@@ -0,0 +1,275 @@
+import {
+  EuiSpacer,
+  EuiFormRow,
+  EuiFieldText,
+  EuiLink,
+  EuiTextArea,
+  EuiButtonEmpty,
+  EuiButton,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiButtonIcon,
+  EuiIcon,
+  EuiText,
+  EuiSuperDatePicker,
+} from '@elastic/eui';
+import { htmlIdGenerator } from '@elastic/eui/lib/services';
+import _ from 'lodash';
+import React, { Fragment, useEffect, useState } from 'react';
+import { VisualizationType } from '../../../../common/constants/custom_panels';
+import PPLService from '../../../services/requests/ppl';
+import { PlotSample } from '../helpers/plot_sample';
+import {
+  convertDateTime,
+  getNewVizDimensions,
+  getQueryResponse,
+  isNameValid,
+} from '../helpers/utils';
+
+/*
+ * "AddNewVisualizations" component to add new visualizations using PPL Queries
+ *
+ * closeVizWindow: function to close "add visualization" window
+ * pplService: PPLService Requestor 
+ * panelVisualizations: panelVisualizations object 
+ * setPanelVisualizations: Setter for panelVisualizations object
+ * setToast: Create Toast function 
+ */
+
+type Props = {
+  closeVizWindow: () => void;
+  pplService: PPLService;
+  panelVisualizations: VisualizationType[];
+  setPanelVisualizations: React.Dispatch<React.SetStateAction<VisualizationType[]>>;
+  setToast: (title: string, color?: string, text?: string) => void;
```

Comment:
FYI might be better to use `notifications` in `CoreStart` if possible, so that core dashboards handles toasts

Reply:
The toasts still works without `notifications` being added. Could you please explain a bit more on how adding it would help? If you could point me to a doc that would be great.

Reply:
Not sure if there are docs, I'm referring to [these functions](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/core/public/notifications/toasts/toasts_api.tsx#L172-L262) from `coreStart.notifications.toasts`.

This makes toasts less flexible since we are not defining the `EuiToast` component, but I can see some benefits:
1. we don't have to maintain toast states and the separate toast component here:
https://github.com/opensearch-project/trace-analytics/blob/d6b3bc98ab333f5e5346ca366d9fb5e4f360c2c1/public/components/custom_panels/home.tsx#L200-L206
2. using `addDanger`, `addSuccess` and other functions takes care of toast color and icon, making it more uniform with Dashboards core UI
3. if there is a toast sent by core or other plugins, toasts sent by `custom_panels` will not be placed on top and cover it up
4. toasts will not disappear if user navigates away from `custom_panels` to other pages like event_explorer or trace_analytics, this can be good or bad depending on the use case

The current implementation also works, so this is just a FYI note

Reply:
I was thinking of a common toast function implementation for all observability modules for long term. The functions form `coreStart.notifications.toasts` look great  and they solve common toast issue. In pt.3 by cover it up, if you mean overlap of toasts then we should implement it using  `notifications`.  But, pt.4 is a bit tricky and may be UX team can help us more on it. For now let's keep it as is. 

If you are starting to implement toasts for trace analytics, we can probably have a second opinion from the UX team, that would save you some double effort.

Reply:
Trace analytics is read only, so I don't think toasts were needed. For point 3 yes, I have not tested it myself but I saw this behavior in reporting (reporting doesn't have access to `coreStart`): if there's an existing toast from core and reporting sends a toast, user cannot see the toast from core unless they close the reporting toast. we can check with UX team on whether we should persist toasts across pages